napari/_qt/dialogs/qt_reader_dialog.py

```diff
@@ -250,7 +255,9 @@ def open_with_dialog_choices(
     qt_viewer.viewer.open(paths, stack=stack, plugin=plugin_name, **kwargs)
 
     if persist:
+        if not os.path.isdir(extension):
+            extension = '*' + extension
```

Comment:
could you explain what is idea for this? 
It is possible that the path to the directory from external drive could be removed in the meantime, so `if` will be satisfied (as `isdir` requires the directory to be accessible during check)?

Reply:
The extension fed into this function does not have an '*' attached to it for a file, but we want that in preferences.  For folders, we do not want the '*'.  I have moved this around a bit, however, one of the tests was failing, so I changed it back to pass.

Reply:
i just tried checking this a different way.  I'm not sure if this helps to solve the problem mentioned above with hte directory being removed and therefore making the if statement pass.

Reply:
@ppwadhwa that looks fine, just need to use `os.sep` in `test_reader_dir` to avoid the windows failures so 

`assert widg._persist_text == f'Remember this choice for folders labeled as {dir}{os.sep}.'`